tests/DCPS/DomainRange/DomainRangeTest.cpp

```diff
@@ -0,0 +1,398 @@
+    DDS::StatusCondition_var condition = writer_->get_statuscondition();
+    condition->set_enabled_statuses(DDS::PUBLICATION_MATCHED_STATUS);
+
+    DDS::WaitSet_var ws = new DDS::WaitSet;
+    ws->attach_condition(condition);
+
+    while (true) {
+      DDS::PublicationMatchedStatus matches;
+      if (writer_->get_publication_matched_status(matches) != ::DDS::RETCODE_OK) {
+        ACE_ERROR_RETURN((LM_ERROR,
+                          ACE_TEXT("ERROR: %N:%l: main() -")
+                          ACE_TEXT(" get_publication_matched_status failed!\n")),
+                         -1);
+      }
+
+      ACE_DEBUG((LM_DEBUG, "(%P|%t) DataWriter %C has %d of %d readers\n", writer_id_.c_str(), matches.current_count, total_readers_));
+      if (matches.current_count >= total_readers_) {
+        break;
+      }
+
+      DDS::ConditionSeq conditions;
+      DDS::Duration_t timeout = { 60, 0 };
+      if (ws->wait(conditions, timeout) != DDS::RETCODE_OK) {
+        ACE_ERROR_RETURN((LM_ERROR,
+                          ACE_TEXT("ERROR: %N:%l: main() -")
+                          ACE_TEXT(" wait failed!\n")),
+                         -1);
+      }
+    }
+
+    ws->detach_condition(condition);
+
+    // Write samples
+    Messenger::Message message;
+    message.from = writer_id_.c_str();
+    message.subject = "Test Message";
+    message.subject_id = 1;
+    message.text = "Testing...";
+    message.count = 0;
+    for (int i = 0; i < MSGS_PER_WRITER; ++i) {
+      DDS::ReturnCode_t error = message_writer->write(message, DDS::HANDLE_NIL);
+      ++message.count;
+
+      if (error != DDS::RETCODE_OK) {
+        ACE_ERROR((LM_ERROR,
+                   ACE_TEXT("ERROR: %N:%l: main() -")
+                   ACE_TEXT(" write returned %d!\n"), error));
+      }
+    }
+
+    ACE_DEBUG((LM_DEBUG, "(%P|%t) DataWriter %C is waiting for acknowledgments\n", writer_id_.c_str()));
+    DDS::Duration_t timeout = { 30, 0 };
+    message_writer->wait_for_acknowledgments(timeout);
+    // With static discovery, it's not an error for wait_for_acks to fail
+    // since the peer process may have terminated before sending acks.
+    ACE_DEBUG((LM_DEBUG, "(%P|%t) DataWriter %C is done\n", writer_id_.c_str()));
+
+    return 0;
+  }
+
+private:
+  std::string writer_id_;
+  DDS::DataWriter_var writer_;
+  int total_readers_;
+};
+
+ACE_Thread_Mutex readers_done_lock;
+ACE_Condition_Thread_Mutex readers_done_cond(readers_done_lock);
+int readers_done = 0;
+
+void reader_done_callback()
+{
+  ACE_Guard<ACE_Thread_Mutex> g(readers_done_lock);
+  ++readers_done;
+  readers_done_cond.signal();
+}
+
+int
+ACE_TMAIN(int argc, ACE_TCHAR *argv[])
+{
+  std::vector<DDS::DomainId_t> domains;
+
+  try {
+    // Initialize DomainParticipantFactory
+    DDS::DomainParticipantFactory_var dpf =
+      TheParticipantFactoryWithArgs(argc, argv);
+
+    const std::string reader_id = "Reader";
+    const std::string writer_id = "Writer";
+    bool reliable = true;
```

Comment:
Reliable is always true, yes? You should either make this const or remove it entirely and remove / fix all the conditionals that use it.

Reply:
Modified the test.